Description: Stampfile handling
 This patch adds support for "stampfile" in config file, as described in
 manual page.
 .
 Originally appeared in version 1.3.0-5.  Converted to quilt for version
 1.3.0+20120605-1.
Author: Juan Cespedes <cespedes@debian.org>
Bug-Debian: https://bugs.debian.org/cgi-bin/bugreport.cgi?bug=97509
Last-Update: 2014-09-11

--- a/wmppp/wmppp.c
+++ b/wmppp/wmppp.c
@@ -192,7 +192,7 @@
 #define SPEED_ACTION (NULL)
 #define IFDOWN_ACTION "/usr/bin/poff -r"
 
-#define STAMP_FILE "/var/run/ppp0.pid"
+#define STAMP_FILE_PRE "/var/run/wmppp."
 
 /* Defines voor alle coordinate */
 
@@ -362,6 +362,7 @@
 char	*stop_action = NULL;
 char	*speed_action = NULL;
 char	*ifdown_action = NULL;
+char    *stamp_file = NULL;
 
 void wmppp_routine(int argc, char **argv) {
 
@@ -370,6 +371,7 @@
 		{ "stop", &stop_action },
 		{ "speed", &speed_action },
 		{ "ifdown", &ifdown_action },
+		{ "stampfile", &stamp_file },
 		{ NULL, NULL }
 	};
 
@@ -411,6 +413,10 @@
 	if (STOP_ACTION) stop_action = strdup(STOP_ACTION);
 	if (SPEED_ACTION) speed_action = strdup(SPEED_ACTION);
 	if (IFDOWN_ACTION) ifdown_action = strdup(IFDOWN_ACTION);
+	if (STAMP_FILE_PRE) {
+	    sprintf (temp, "%s%s", STAMP_FILE_PRE, active_interface);
+	    stamp_file = strdup (temp);
+	}
 	
 	strcpy(temp, "/etc/wmppprc");
 	parse_rcfile(temp, wmppp_keys);
@@ -483,7 +489,7 @@
 				if (!starttime) {
 					starttime = currenttime;
 
-					if (stat(STAMP_FILE, &st) == 0)
+					if (stat(stamp_file, &st) == 0)
 						starttime = st.st_mtime;
 
 					SetOnLED(LED_PPP_POWER);
